android/app/src/main/java/app/organicmaps/MwmActivity.java

```diff
@@ -1937,8 +1934,13 @@ private void onLocationPermissionsResult(java.util.Map<String, Boolean> permissi
       if (LocationState.getMode() == LocationState.NOT_FOLLOW_NO_POSITION)
         LocationState.nativeSwitchToNextMode();
 
-      if (requestedForRecording && LocationUtils.checkFineLocationPermission(this))
-        startTrackRecording(true);
+      if (requestedForRecording)
+      {
+        if (LocationUtils.checkFineLocationPermission(this))
+          startTrackRecording(true);
+        else if (TrackRecorder.nativeIsEnabled())
```

Comment:
Should any message be displayed to a user in such case? How to reproduce it in tests?

Reply:
Steps to regenrate

1. Turn on trace path feature
2. Close the app and clear it from recent apps
3. Revoke the location permissions for OM
4. Restart the OM
5. OM will ask for location permission deny it
6. Trace path feature will again try to get the location permission if user denies it then native portion of track recorder will turned off (i.e. on restart of track recording, recorded tracks on map will get deleted)